DEF-2736 Editor support for texture profiles

ragnard-king opened the pull request:
Fixes https://github.com/defold/editor2-issues/issues/1292

### User facing changes

- Texture profiles are now respected both when viewing textures in editor, and when starting application using "Project -> Build".

### Technical changes

- support for texture profiles
  - used when generating texture in editor
  - used when building, optionally with compression
- new `build-settings` property on Workspace, set from prefs when building
- new `texture-profiles` property on Project
- updated `TexcLibrary` and `TextureGenerator` from bob
- new function go generate gpu-texture using `TextureImage` instead of `BufferedImage`, ie. an image that is generate from our pipeline

ragnard-king committed:
DEF-2736 Editor2 support for texture profiles

erikangelin-king reviewed (changes requested):
Minor remarks, looks good!

ragnard-king committed:
DEF-2736 Editor2 review fixes

- move update of workspace :build-settings property to app-view
- remove some unused output arguments
- add type hint to texture-image->texture-data

ragnard-king committed:
DEF-2736 Review fixes

- set workspace :build-settings property when creating workspace, and
  when changing preferences
